generated/language/src/main/java/com/google/cloud/language/v1/spring/LanguageServiceSpringAutoConfiguration.java

```diff
@@ -0,0 +1,471 @@
+    }
+    return ((CredentialsProvider) new DefaultCredentialsProvider(this.globalProperties));
+  }
+
+  /**
+   * Returns the default channel provider. The default is gRPC and will default to it unless the
+   * useRest option is provided to use HTTP transport instead
+   */
+  @Bean
+  @ConditionalOnMissingBean
+  public TransportChannelProvider defaultLanguageServiceTransportChannelProvider() {
+    return LanguageServiceSettings.defaultTransportChannelProvider();
+  }
+
+  /**
+   * Provides a LanguageServiceClient bean configured to use the default credentials provider
+   * (obtained with languageServiceCredentials()) and its default transport channel provider
+   * (defaultLanguageServiceTransportChannelProvider()). It also configures the quota project ID if
+   * provided. It will configure an executor provider in case there is more than one thread
+   * configured in the client
+   *
+   * <p>Individual retry settings are configured as well. It will use the relevant client library's
+   * default retry settings when they are not specified in LanguageServiceSpringProperties.
+   */
+  @Bean
+  @ConditionalOnMissingBean
+  public LanguageServiceClient languageServiceClient(
+      @Qualifier("languageServiceCredentials") CredentialsProvider credentialsProvider,
+      @Qualifier("defaultLanguageServiceTransportChannelProvider")
+          TransportChannelProvider defaultTransportChannelProvider)
+      throws IOException {
+    LanguageServiceSettings.Builder clientSettingsBuilder =
+        LanguageServiceSettings.newBuilder()
+            .setCredentialsProvider(credentialsProvider)
+            .setTransportChannelProvider(defaultTransportChannelProvider)
+            .setHeaderProvider(this.userAgentHeaderProvider());
+    if (this.clientProperties.getQuotaProjectId() != null) {
+      clientSettingsBuilder.setQuotaProjectId(this.clientProperties.getQuotaProjectId());
+      LOGGER.info(
+          "Quota project id set to "
+              + this.clientProperties.getQuotaProjectId()
+              + ", this overrides project id from credentials.");
+    }
+    if (this.clientProperties.getExecutorThreadCount() != null) {
+      ExecutorProvider executorProvider =
+          LanguageServiceSettings.defaultExecutorProviderBuilder()
+              .setExecutorThreadCount(this.clientProperties.getExecutorThreadCount())
+              .build();
+      clientSettingsBuilder.setBackgroundExecutorProvider(executorProvider);
+      LOGGER.info(
+          "Background executor thread count is " + this.clientProperties.getExecutorThreadCount());
+    }
+    if (this.clientProperties.getUseRest()) {
+      clientSettingsBuilder.setTransportChannelProvider(
+          LanguageServiceSettings.defaultHttpJsonTransportProviderBuilder().build());
+      LOGGER.info("Using HTTP transport channel");
+    }
+    RetrySettings.Builder analyzeSentimentRetrySettingBuilder =
```

Comment:
`analyzeSentimentRetrySettings` could also be its own bean to allow customization. One thing to remember is that Spring Boot allows you to configure the app via properties, and also by overwriting beans. For the overwriting beans route, you want to make the beans as granular as possible, for maximum flexiblity.

Reply:
Are you suggesting to have an RetrySetting bean for each RPC? If it has additional benefit other than readability, I think we should do it too.

Reply:
Yes. For example, it also allows you to use the same retry settings for all RPCs.